packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -433,6 +467,167 @@ export class GraphQLApi extends Construct {
     });
   }
 
+  /**
+   * Adds an IAM policy statement associated with this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal
+   * @param resources The set of resources to allow (i.e. ...:[region]:[accountId]:apis/GraphQLId/...)
+   * @param actions The actions that should be granted to the principal (i.e. appsync:graphql )
+   */
+  public grant(grantee: IGrantable, resources: IamResources[], ...actions: string[]): Grant {
+    return Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: resources.map((resource) => {
+        return this.generateResourceArn(resource);
+      }),
+      scope: this,
+    });
+
+  }
+
+  /**
+   * Adds an IAM policy statement for full access to this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal
+   */
+  public grantFullAccess(grantee: IGrantable): Grant {
+    return Grant.addToPrincipal({
+      grantee,
+      actions: ['appsync:GraphQL'],
+      resourceArns: [
+        this.generateResourceArn({custom: '*'}),
+      ],
+      scope: this,
+    });
+  }
+
+  /**
+   * Adds an IAM policy statement for Mutation access to this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal
+   * @param fields The fields to grant access to that are Mutations (leave blank for all)
+   */
+  public grantMutation(grantee: IGrantable, fields?: string[]): Grant {
+    return this.grantType(grantee, 'Mutation', fields);
+  }
+
+  /**
+   * Adds an IAM policy statement for Query access to this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal
+   * @param fields The fields to grant access to that are Queries (leave blank for all)
+   */
+  public grantQuery(grantee: IGrantable, fields?: string[]): Grant {
+    return this.grantType(grantee, 'Query', fields);
+  }
+
+  /**
+   * Adds an IAM policy statement for Subscription access to this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal
+   * @param fields The fields to grant access to that are Subscriptions (leave blank for all)
+   */
+  public grantSubscription(grantee: IGrantable, fields?: string[]): Grant {
+    return this.grantType(grantee, 'Subscription', fields);
+  }
+
+  /**
+   * Adds an IAM policy statement for Type access to this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal
+   * @param type The type to grant access
+   * @param fields The fields to grant access to that are of @param type (leave blank for all)
+   */
+  public grantType(grantee: IGrantable, type: string, fields?: string[]): Grant {
```

Comment:
Can you explain, what is the difference between `grantType(grantee, myType, myFields)` and `grant(grantee, [{ type: myType, field: myFields[0] }, { type: myType, field: myFields[1] }], 'appsync:GraphQL')`?

Reply:
`grantType` is more of a nice to have so that its easier to write out multiple fields of a certain type.

for example if i have something like 
```gql
type specialType {
  secretInfo {
  ...
  }
    @aws_iam
  ...
  otherSecretInfo {
  ...
  }
   @aws-iam
}
```

I was thinking that it would be nice to just do it like `api.grantType(role, 'specialType', [secretInfo, otherSecretInfo])`

`grant` was more of a escape hatch in my mind

Reply:
Isn't it better to change `IamResources` to take multiple `field`s then, instead of just one? I don't think having 2 slightly different ways to achieve the same thing is a good idea.

Reply:
mmm yeah thats a good point i think in that case just removing `grantType` and changing `field` to `fields` is the way to go.